rtengine/iptransform.cc

```diff
@@ -740,83 +849,70 @@ void ImProcFunctions::transformLuminanceOnly (Imagefloat* original, Imagefloat*
 
 void ImProcFunctions::transformGeneral(bool highQuality, Imagefloat *original, Imagefloat *transformed, int cx, int cy, int sx, int sy, int oW, int oH, int fW, int fH, const LensCorrection *pLCPMap)
 {
+    BENCHFUN
     // set up stuff, depending on the mode we are
-    bool enableLCPDist = pLCPMap && params->lensProf.useDist;
-    bool enableCA = highQuality && needsCA();
-    bool enableGradient = needsGradient();
-    bool enablePCVignetting = needsPCVignetting();
-    bool enableVignetting = needsVignetting();
-    bool enablePerspective = needsPerspective();
-    bool enableDistortion = needsDistortion();
+    const bool enableLCPDist = pLCPMap && params->lensProf.useDist;
+    const bool enableCA = highQuality && needsCA();
+    const bool enableGradient = needsGradient();
+    const bool enablePCVignetting = needsPCVignetting();
+    const bool enableVignetting = needsVignetting();
+    const bool enablePerspective = needsPerspective();
+    const bool enableDistortion = needsDistortion();
 
-    double w2 = (double) oW  / 2.0 - 0.5;
-    double h2 = (double) oH  / 2.0 - 0.5;
+    const double w2 = (double) oW  / 2.0 - 0.5;
+    const double h2 = (double) oH  / 2.0 - 0.5;
 
     double vig_w2, vig_h2, maxRadius, v, b, mul;
-    calcVignettingParams (oW, oH, params->vignetting, vig_w2, vig_h2, maxRadius, v, b, mul);
+    calcVignettingParams(oW, oH, params->vignetting, vig_w2, vig_h2, maxRadius, v, b, mul);
 
     struct grad_params gp;
 
     if (enableGradient) {
-        calcGradientParams (oW, oH, params->gradient, gp);
+        calcGradientParams(oW, oH, params->gradient, gp);
     }
 
     struct pcv_params pcv;
 
     if (enablePCVignetting) {
-        calcPCVignetteParams (fW, fH, oW, oH, params->pcvignette, params->crop, pcv);
+        calcPCVignetteParams(fW, fH, oW, oH, params->pcvignette, params->crop, pcv);
     }
 
-    float** chOrig[3];
-    chOrig[0] = original->r.ptrs;
-    chOrig[1] = original->g.ptrs;
-    chOrig[2] = original->b.ptrs;
-
-    float** chTrans[3];
-    chTrans[0] = transformed->r.ptrs;
-    chTrans[1] = transformed->g.ptrs;
-    chTrans[2] = transformed->b.ptrs;
+    float** chOrig[3] = {original->r.ptrs, original->g.ptrs, original->b.ptrs};
```

Comment:
`const float* const * const` doesn't really look good, so I'll try to make use of std::array<>.